beacon-chain/attestation/service.go

```diff
@@ -246,59 +249,29 @@ func (a *Service) updateAttestation(ctx context.Context, headRoot [32]byte, beac
 	attestation *pb.Attestation) error {
 	totalAttestationSeen.Inc()
 
-	slot := attestation.Data.Slot
-	var committee []uint64
-	var cachedCommittees *cache.CommitteesInSlot
-	var err error
-
-	for beaconState.Slot < slot {
-		beaconState, err = state.ExecuteStateTransition(
-			ctx, beaconState, nil /* block */, headRoot, state.DefaultConfig(),
-		)
-		if err != nil {
-			return fmt.Errorf("could not execute head transition: %v", err)
-		}
-	}
-
-	cachedCommittees, err = committeeCache.CommitteesInfoBySlot(slot)
+	committee, err := a.committee(ctx, headRoot, beaconState, attestation)
 	if err != nil {
 		return err
 	}
-	if cachedCommittees == nil {
-		crosslinkCommittees, err := helpers.CrosslinkCommitteesAtSlot(beaconState, slot, false /* registryChange */)
-		if err != nil {
-			return err
-		}
-		cachedCommittees = helpers.ToCommitteeCache(slot, crosslinkCommittees)
-		if err := committeeCache.AddCommittees(cachedCommittees); err != nil {
-			return err
-		}
-	}
 
-	// Find committee for shard.
-	for _, v := range cachedCommittees.Committees {
-		if v.Shard == attestation.Data.Shard {
-			committee = v.Committee
-			break
+	bitfield := attestation.AggregationBitfield
+	if !a.canProcessBitfield(committee, bitfield, len(beaconState.ValidatorRegistry)) {
+		if featureconfig.FeatureConfig().EnableForkedAttestationProcessing {
+			a.forkedAttestations = append(a.forkedAttestations, attestation)
 		}
+
+		return fmt.Errorf("invalid state to process forked attestation at slot %d",
```

Comment:
you should make log this as a debug , and then return it

Reply:
got it, thanks for the feedback!